### PR TITLE
Multiple File Uploader

### LINKED ISSUE #2496: Multiple File Uploader
https://github.com/ProcessMaker/screen-builder/issues/420

### PR DESCRIPTION
**Pull request by marcoAntonioNina**
closes #2496 

## Changes
- Add checkbox for enable multiple files.
- Allow upload multiples files by request.

## Test
- create form with upload file and enable multiple files.
- create a proces and task with form.
- the form allow upload multiple files by request.

## Screenshot
image

image

### CODE REVIEW

**Comment by marcoAntonioNina** [comment]
@nolanpro Test added.

File: ProcessMaker/Http/Controllers/Api/ProcessRequestFileController.php [hunk 1]
```diff
@@ -86,26 +86,26 @@ class ProcessRequestFileController extends Controller
...
-			return new ResourceCollection($media);
-		} else {
-			$filtered = $media->reject(function ($item, $key) use ($name) {
-				if ($item->custom_properties['data_name'] != $name) {
-					return true;
-				}
-			});
+        //If no name, return entire collection; otherwise, filter collection
+        if (!$name) {
+            return new ResourceCollection($media);
```
Comment (danloa) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestFileController.php]: This block looks like a  guard condition that exits the function returning a ne ResourceCollection. So the else part is not necessary. If you do this, the code nesting will diminish and the code will be more readble. For example:
if(!name) {
    return ResourceCollection($media);
}

$filtered = $media....
:
return new ResourceCollection($filtered)

File: ProcessMaker/Http/Controllers/Api/ProcessRequestFileController.php [hunk 2]
```diff
@@ -158,46 +158,54 @@ public function show(Request $laravel_request, ProcessRequest $request, Media $f
...
-            return response()->json([
-                "done" => $handler->getPercentageDone()
-            ]);
+
+            // save the file and return any response you need
+            $file = $request
+                ->addMedia($save->getFile())
+                ->withCustomProperties(['data_name' => $data_name]) // photo_1
+                ->toMediaCollection();
+            // $identifier = ['_type' => 'file', 'id' => $file->id];
```
Comment (danloa) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestFileController.php]: Remove the commented code

File: ProcessMaker/Http/Controllers/Api/ProcessRequestFileController.php [hunk 3]
```diff
@@ -158,46 +158,54 @@ public function show(Request $laravel_request, ProcessRequest $request, Media $f
...
+            return new JsonResponse(['message' => 'The file was uploaded.', 'fileUploadId' => $file->id], 200);
+        }
+        // we are in chunk mode, lets send the current progress
+        /** @var AbstractHandler $handler */
```
Comment (danloa) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestFileController.php]: Can you explain better this comment?, or remove it

**Comment by marcoAntonioNina** [comment]
@danloa fix comments

**Pull request closed without merging**